dotnet-client-libraries/BenchmarkDotNetWithUnitTests/BenchmarkDotNetWithUnitTests/StringConcatBenchmarks.cs

```diff
@@ -0,0 +1,56 @@
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Order;
+
+namespace BenchmarkDotNetWithUnitTests;
+
+[MemoryDiagnoser]
+[Orderer(SummaryOrderPolicy.FastestToSlowest)]
+[RankColumn]
+public class StringConcatBenchmarks
+{
+    private string? _firstName;
+    private string? _lastName;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        _firstName = "John";
+        _lastName = "Doe";
+    }
+
+    [Benchmark]
+    public string StringConcat()
+    {
+        return _firstName + " " + _lastName;
+    }
+
+    [Benchmark]
+    public string StringFormat()
+    {
+        return string.Format("{0} {1}", _firstName, _lastName);
+    }
+
+    [Benchmark]
+    public string StringInterpolation()
+    {
+        return $"{_firstName} {_lastName}";
+    }
+
+    [Benchmark]
+    public string StringConcatWithJoin()
+    {
+        return string.Join(" ", _firstName, _lastName);
+    }
+
+    [Benchmark]
+    public string StringConcatWithJoinAndArray()
+    {
+        return string.Join(" ", _firstName, _lastName);
+    }
+
+    [Benchmark]
+    public string StringConcatWithJoinAndParams()
+    {
+        return string.Join(" ", _firstName, _lastName);
+    }
```

Comment:
All of these methods are calling into the same overload of `string.Join`. If you exam the lowered code in something like Sharplab.io you will see that behind the scenes the compiler has converted each of these methods into identical code where the two strings are loaded into a new array of string and then the string.Join overload that takes an array of strings is being called. You can also look at the compiled IL and see the same thing happening, it is just easier to read in the lowered code 😀